### ISSUE TITLE
Ignore generated ral code

### ISSUE THREAD

**Issue by bfrog** [imxrt-rs/imxrt-rs#15]
I think it might make sense to ignore any generated code, to make it clearer that those files really shouldn't be edited by hand. It does mean any contributor would need to have python and some tooling installed (though already done semi-automatically via make venv) to generate the ral files.

I'd like some feedback on that before doing it.

**Comment by bfrog** [comment]
I went ahead and did it, but will wait on merging based on feedback. I'm also looking to automatically get a CI flow going in github actions that generates the code and builds it at the very least.
